dockerfiles/QAT/Dockerfile

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Kata osbuilder 'works best' on Fedora
+FROM fedora:latest
+
+# Version of the Dockerfile
+LABEL DOCKERFILE_VERSION="1.0"
+
+ENV QAT_DRIVER_VER qat1.7.l.4.9.0-00008.tar.gz
+ENV QAT_DRIVER_URL https://01.org/sites/default/files/downloads/${QAT_DRIVER_VER}
+ENV QAT_COMPILE_OPTIONS --disable-qat-lkcf --enable-icp-sriov=guest
+
+ENV KATA_VERSION master
+ENV OUTPUT_DIR /output
+
+RUN dnf install -y \
+    bc \
+    bison \
+    curl \
+    diffutils \
+    e2fsprogs \
+    elfutils-libelf-devel \
+    findutils \
+    flex \
+    gcc \
+    gcc-c++ \
+    git \
+    kmod \
+    openssl-devel \
+    make \
+    parted \
+    patch \
+    qemu-img \
+    systemd-devel \
+    sudo \
+    xz
+
+# Pull in our local files
+COPY ./run.sh /input/
```

Comment:
I had to manually `chmod +x` this before building the container image when I downloaded the file from the GitHub raw using curl.

Reply:
that's because your `curl` dropped the execute bit from the script. Don't do that - that is not how you get code from github ;-) - you need to `git clone` the repo :-)

Reply:
Yes it took me a bit of time to figure that out. This is the first time I had run into this particular issue because I always curl Dockerfile's from github directly. Maybe that is the wrong way to do it, but I have seen instructions throughout the years that do just that.  There are others who are like me that do things perhaps a non-standard way and we have to design for that.  That's why I am here to test :)    Couldn't you just git clone this osbuilder repo as part of the Dockerfile and then add in the additional files then?
```
RUN git clone <osbuilder repo>
```
That would eliminate the need to download multiple files to the same folder, would eliminate the possibility someone curl's a file and loses the execute permission, and would allow the Dockerfile to just be emailed to people and work. It might be a little tricky to test before committing all the files, but it would make things easier on the user.  You could just email the Dockerfile around and that would be all you need. Also, instead of trying to put all the complex logic of `run.sh` into a Dockerfile to eliminate the need of that file you could just clone all the necessary scripts needed to build the QAT container.

Reply:
I'm going to say nack to the idea of 'just one dockerfile', for a number of reasons:
 - nobody is going to `curl` down just the dockerfile and run it - **unless** we tell them to do that - and, we don't. They will only find the Dockerfile and docs by visiting the repo and reading the instructions, which do not say `curl` down one file and run it... I'll make it even clearer in the documentation you need to git clone (or at least have the whole QAT directory) present. You happen to have curl'd the file down on its own because you only read a fragment of the PR. once the PR is merged, nobody will be doing that, they will start from the markdown instructions.
 - wiring those git urls into the Dockerfile pins the dockerfile to precisely that version (master) of what is on github, and to that path (url). **not** wiring those into the Dockerfile means the `docker build` will work wherever you place the files (no hard wired paths), and it allows you to edit your local version and build it. Both good things. Let's not wire hard paths into the Dockerfile.
 - putting that git clone into the Dockerfile, as above, would mean the git repo would end up in the docker image as well. There are ways to mitigate that (both with a shallow clone and 're-basing' the Dockerfile after the clone and copy), but, really, this is all adding un-necessary complexity. As it stands, the Dockerfile and additional COPY files fit pretty well with the KISS principle and are standard Dockerfile practices. Let's keep it that way.

Reply:
Fair enough. My goal is to make this as easy as possible for end users even if it is a bit more complex to maintain.